assets/src/edit-story/components/canvas/carousel/index.js

```diff
@@ -91,6 +95,21 @@ function Carousel() {
 		}
 	}, [ currentPageId, hasHorizontalOverflow, pageRefs ] );
 
+	useLayoutEffect( () => {
+		const listElement = listRef.current;
+
+		const handleScroll = () => {
+			const max = listElement.scrollWidth - listElement.offsetWidth;
+			setScrollPercentage( listElement.scrollLeft / max );
+		};
+
+		listElement.addEventListener( 'scroll', handleScroll, { passive: true } );
```

Comment:
Just in case: is the support for `passive` acceptable? I suppose so.

Reply:
Support seems to be quite good! https://caniuse.com/#feat=passive-event-listener